lib/fixtury/definition_executor.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module Fixtury
+  class DefinitionExecutor
+
+    attr_reader :value, :execution_type, :definition, :store, :execution_context
+
+    def initialize(store: nil, execution_context: nil, definition:)
+      @store = store
+      @execution_context = execution_context || self
+      @definition = definition
+      @execution_type = nil
+      @value = nil
+    end
+
+    def __call
+      maybe_set_store_context do
+        run_callable(callable: definition.callable, type: :definition)
+        definition.enhancements.each do |e|
+          run_callable(callable: e, type: :enhancement)
+        end
+      end
+      value
+    end
+
+    def get(*args)
+      raise ArgumentError, "A store is required for #{definition.name}" unless store
+
+      if execution_context&.respond_to?(:around_fixture_get)
```

Comment:
think we should have before and after too?

Reply:
I'd vote no. I'm not sure what you'd ever do before/after that you can't implement in an around. If you're looking for consistency between this and the other hooks, I'd be more inclined to remove the before/after in the fixture hooks